test/e2e/cmd/run/diagnostics.go

```diff
@@ -0,0 +1,130 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package run
+
+import (
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"path/filepath"
+	"sync"
+	"time"
+
+	v1 "github.com/elastic/cloud-on-k8s/pkg/apis/elasticsearch/v1"
+	"github.com/elastic/cloud-on-k8s/test/e2e/test"
+)
+
+type diagnosticContext struct {
+	test.Context
+	ESNamespace string
+	ESName      string
+	PodName     string
+	TLS         bool
+}
+
+func (h *helper) runEsDiagnosticsJob() {
+	output, err := h.kubectl("get", "es", "--all-namespaces", "-o", "json")
+	if err != nil {
+		log.Error(err, "failed to list Elasticsearch clusters")
+		return
+	}
+	var ess v1.ElasticsearchList
+	if err := json.Unmarshal([]byte(output), &ess); err != nil {
+		log.Error(err, "failed to unmarshal kubectl response")
+	}
+
+	if len(ess.Items) == 0 {
+		log.Info("No Elasticsearch clusters deployed. Cannot extract diagnostics")
+		return
+	}
+
+	var wg sync.WaitGroup
+	wg.Add(len(ess.Items))
+
+	diagnosticsTimeout := 10 * time.Minute
+
+	for _, es := range ess.Items {
+		go func(es v1.Elasticsearch) {
+			defer wg.Done()
+			podName := fmt.Sprintf("diag-%s", es.Name)
+			err := h.kubectlApplyTemplateWithCleanup("config/e2e/diagnostics_job.yaml", diagnosticContext{
```

Comment:
nit: there are a few places where the returned values can be scoped to the `if` block that handles them:

```golang
			if err := h.kubectlApplyTemplateWithCleanup("config/e2e/diagnostics_job.yaml", diagnosticContext{
				Context:     h.testContext,
				ESNamespace: es.Namespace,
				ESName:      es.Name,
				PodName:     podName,
				TLS:         es.Spec.HTTP.TLS.Enabled(),
			}); err != nil {
				log.Error(err, "diagnostics pod failed to create")
				return
			}
```

Reply:
Sometimes I find it it easier to parse visually if it is a separate statement, especially if the call producing the value is already spreading over multiple lines like here. But I changed it and the others too.